### PR TITLE
Implement Phase 2: SyscallAnalyzer with backward scan

### PR DESCRIPTION
**Pull request by isseis**
- Add SyscallAnalyzer with AnalyzeSyscallsFromELF() for direct syscall instruction detection (Pass 1) and Go wrapper call analysis skeleton (Pass 2)
- Implement backward scan algorithm with windowed decoding for efficient syscall number extraction from x86_64 machine code
- Add GoWrapperResolver stub (full implementation in Phase 3)
- Add comprehensive tests covering immediate/indirect/control flow boundary/scan limit scenarios (12 test functions)
- Update implementation plan: mark Phase 2 items as complete

### CODE REVIEW

File: internal/runner/security/elfanalyzer/syscall_analyzer.go [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+	pos := startOffset
+
+	for pos < endOffset {
+		// Slice input to [pos:endOffset] to prevent decoding beyond window boundary.
+		// This ensures the decoder cannot consume bytes past endOffset (e.g., the syscall instruction itself).
+		// Validate pos is non-negative before converting to uint64 to prevent overflow.
+		if pos < 0 {
+			break
+		}
+		inst, err := a.decoder.Decode(code[pos:endOffset], baseAddr+uint64(pos))
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/syscall_analyzer.go]: done

File: internal/runner/security/elfanalyzer/syscall_analyzer.go [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+	return &SyscallAnalyzer{
+		decoder:         NewX86Decoder(),
+		syscallTable:    NewX86_64SyscallTable(),
+		goResolver:      NewGoWrapperResolver(),
+		maxBackwardScan: defaultMaxBackwardScan,
+	}
+}
+
+// NewSyscallAnalyzerWithConfig creates a SyscallAnalyzer with custom configuration.
+func NewSyscallAnalyzerWithConfig(decoder MachineCodeDecoder, table SyscallNumberTable, maxScan int) *SyscallAnalyzer {
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/syscall_analyzer.go]: done

File: internal/runner/security/elfanalyzer/syscall_analyzer.go [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+	pos := startOffset
+
+	for pos < endOffset {
+		// Slice input to [pos:endOffset] to prevent decoding beyond window boundary.
+		// This ensures the decoder cannot consume bytes past endOffset (e.g., the syscall instruction itself).
+		// Validate pos is non-negative before converting to uint64 to prevent overflow.
+		if pos < 0 {
+			break
+		}
+		inst, err := a.decoder.Decode(code[pos:endOffset], baseAddr+uint64(pos)) //nolint:gosec
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/syscall_analyzer.go]: fixed

File: internal/runner/security/elfanalyzer/go_wrapper_resolver.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+func (r *GoWrapperResolver) LoadSymbols(elfFile *elf.File) error {
+	// Full implementation in Phase 3 (pclntab_parser.go + go_wrapper_resolver.go)
+	_ = elfFile
+	return nil
+}
+
+// HasSymbols returns true if symbols were successfully loaded.
+func (r *GoWrapperResolver) HasSymbols() bool {
+	return r.hasSymbols
+}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/go_wrapper_resolver.go]: done

File: internal/runner/security/elfanalyzer/syscall_analyzer.go [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+
+		// Check if this instruction modifies eax/rax
+		if !a.decoder.ModifiesEAXorRAX(inst) {
+			continue
+		}
+
+		// Check if it's an immediate move
+		if isImm, value := a.decoder.IsImmediateMove(inst); isImm {
+			return int(value), DeterminationMethodImmediate
+		}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/security/elfanalyzer/syscall_analyzer.go]: done
